File: frontend/src/constants/equipmentCatalog.ts
/**
 * Individual equipment the user can pick per plan. Keys mirror
 * backend/app/services/equipment_catalog.py — keep them in sync.
 */
export const EQUIPMENT_GROUPS: { title: string; items: { value: string; label: string }[] }[] = [
  {
    title: 'Free weights',
    items: [
      { value: 'barbell', label: 'Barbell' },
      { value: 'dumbbells', label: 'Dumbbells' },
      { value: 'kettlebells', label: 'Kettlebells' },
      { value: 'ez_bar', label: 'EZ curl bar' },
    ],
  },
  {
    title: 'Racks & benches',
    items: [
      { value: 'squat_rack', label: 'Squat / power rack' },
      { value: 'bench', label: 'Bench' },
      { value: 'pull_up_bar', label: 'Pull-up bar' },
      { value: 'dip_station', label: 'Dip station' },
      { value: 'plyo_box', label: 'Plyo box' },
    ],
  },
  {
    title: 'Machines',
    items: [
      { value: 'smith_machine', label: 'Smith machine' },
      { value: 'leg_press', label: 'Leg press' },
      { value: 'hack_squat', label: 'Hack squat' },
      { value: 'leg_extension', label: 'Leg extension' },
      { value: 'leg_curl', label: 'Leg curl' },
      { value: 'chest_press_machine', label: 'Chest press' },
      { value: 'pec_deck', label: 'Pec deck / fly' },
      { value: 'shoulder_press_machine', label: 'Shoulder press' },
      { value: 'lat_pulldown', label: 'Lat pulldown' },
      { value: 'seated_row', label: 'Seated cable row' },
      { value: 'cable_crossover', label: 'Cable crossover' },
      { value: 'calf_raise_machine', label: 'Calf raise' },
      { value: 'hip_thrust_machine', label: 'Hip thrust / glute' },
      { value: 'assisted_pull_up', label: 'Assisted pull-up' },
    ],
  },
  {
    title: 'Cardio',
    items: [
      { value: 'treadmill', label: 'Treadmill' },
      { value: 'stationary_bike', label: 'Bike' },
      { value: 'rowing_machine', label: 'Rower' },
      { value: 'elliptical', label: 'Elliptical' },
      { value: 'stair_climber', label: 'Stair climber' },
    ],
  },
  {
    title: 'Other',
    items: [
      { value: 'bands', label: 'Resistance bands' },
      { value: 'medicine_ball', label: 'Medicine ball' },
      { value: 'stability_ball', label: 'Stability ball' },
      { value: 'suspension_trainer', label: 'TRX / suspension' },
      { value: 'battle_ropes', label: 'Battle ropes' },
    ],
  },
];

export const ALL_EQUIPMENT = EQUIPMENT_GROUPS.flatMap((g) => g.items.map((i) => i.value));

/** Onboarding's broad choices → sensible individual defaults, for users who haven't picked per-plan yet. */
const FROM_PROFILE: Record<string, string[]> = {
  barbell: ['barbell', 'squat_rack', 'bench', 'ez_bar'],
  dumbbell: ['dumbbells', 'bench'],
  machine: [
    'smith_machine', 'leg_press', 'leg_extension', 'leg_curl', 'chest_press_machine', 'pec_deck',
    'lat_pulldown', 'seated_row', 'cable_crossover', 'calf_raise_machine',
  ],
  bodyweight: ['pull_up_bar', 'dip_station'],
  kettlebell: ['kettlebells'],
  bands: ['bands'],
};

/** A typical commercial gym — used when the profile says nothing about equipment. */
const STANDARD_GYM = [
  'barbell', 'dumbbells', 'ez_bar', 'squat_rack', 'bench', 'pull_up_bar', 'dip_station', 'smith_machine',
  'leg_press', 'leg_extension', 'leg_curl', 'chest_press_machine', 'pec_deck', 'lat_pulldown', 'seated_row',
  'cable_crossover', 'calf_raise_machine', 'treadmill', 'stationary_bike', 'rowing_machine',
];

export function defaultEquipmentFor(profileEquipment: string[] | null | undefined): string[] {
  if (!profileEquipment?.length) return STANDARD_GYM;
  const picked = new Set(profileEquipment.flatMap((e) => FROM_PROFILE[e] ?? []));
  return ALL_EQUIPMENT.filter((e) => picked.has(e));
}

File: backend/app/services/equipment_catalog.py
"""
Individual equipment a user can pick per plan (Plan preferences screen).
Each item maps to a display name for the prompt and to the free-exercise-db
`equipment` values it unlocks when choosing which glossary exercises to offer.
Keep keys in sync with frontend/src/constants/equipmentCatalog.ts.
"""

EQUIPMENT_CATALOG: dict[str, tuple[str, tuple[str, ...]]] = {
    # Free weights
    "barbell": ("Barbell", ("barbell",)),
    "dumbbells": ("Dumbbells", ("dumbbell",)),
    "kettlebells": ("Kettlebells", ("kettlebells",)),
    "ez_bar": ("EZ curl bar", ("e-z curl bar",)),
    # Racks & benches
    "squat_rack": ("Squat / power rack", ()),  # needs a barbell to be useful
    "bench": ("Flat / adjustable bench", ()),  # enables bench variants of whatever weights are picked
    "pull_up_bar": ("Pull-up bar", ("body only",)),
    "dip_station": ("Dip station", ("body only",)),
    "plyo_box": ("Plyo box", ("body only", "other")),
    # Machines
    "smith_machine": ("Smith machine", ("machine",)),
    "leg_press": ("Leg press", ("machine",)),
    "hack_squat": ("Hack squat machine", ("machine",)),
    "leg_extension": ("Leg extension machine", ("machine",)),
    "leg_curl": ("Leg curl machine", ("machine",)),
    "chest_press_machine": ("Chest press machine", ("machine",)),
    "pec_deck": ("Pec deck / fly machine", ("machine",)),
    "shoulder_press_machine": ("Shoulder press machine", ("machine",)),
    "lat_pulldown": ("Lat pulldown", ("cable", "machine")),
    "seated_row": ("Seated cable row", ("cable",)),
    "cable_crossover": ("Cable crossover / functional trainer", ("cable",)),
    "calf_raise_machine": ("Calf raise machine", ("machine",)),
    "hip_thrust_machine": ("Hip thrust / glute machine", ("machine",)),
    "assisted_pull_up": ("Assisted pull-up / dip machine", ("machine",)),
    # Cardio
    "treadmill": ("Treadmill", ("machine", "other")),
    "stationary_bike": ("Stationary bike", ("machine", "other")),
    "rowing_machine": ("Rowing machine", ("machine", "other")),
    "elliptical": ("Elliptical", ("machine", "other")),
    "stair_climber": ("Stair climber", ("machine", "other")),
    # Other
    "bands": ("Resistance bands", ("bands",)),
    "medicine_ball": ("Medicine ball", ("medicine ball",)),
    "stability_ball": ("Stability ball", ("exercise ball",)),
    "suspension_trainer": ("TRX / suspension trainer", ("body only", "other")),
    "battle_ropes": ("Battle ropes", ("other",)),
}

EQUIPMENT_KEYS: tuple[str, ...] = tuple(EQUIPMENT_CATALOG)


def display_names(keys: list[str]) -> list[str]:
    return [EQUIPMENT_CATALOG[k][0] for k in keys if k in EQUIPMENT_CATALOG]


def glossary_equipment(keys: list[str]) -> set[str]:
    """free-exercise-db equipment values unlocked by the picked items (bodyweight always included)."""
    values = {"body only"}
    for key in keys:
        values.update(EQUIPMENT_CATALOG.get(key, ("", ()))[1])
    return values
